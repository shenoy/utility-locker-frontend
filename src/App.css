body {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgb(238, 234, 234);
}

.item {
  color: rgb(1, 1, 20);
  border-radius: 10px;
  border-color: rgb(248, 245, 238);
  text-align: left;
  align-items: center;
  background-color: white;
  border-style: outset;
  padding-left: 30%;
  padding-top: 20px;
  padding-bottom: 20px;
  font-family: "Open Sans Condensed", sans-serif;
  font-weight: 300;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);

}

.flex {
  display: flex;
  align-items: center;
  font-size: 1.5rem;
}

.property {
  color: rgb(32, 70, 238);
  font-size: 1.5rem;
}

.sort,
.group {
  color: rgb(33, 102, 12);
  font-size: 1.5rem;
  border-color: rgb(248, 245, 238);
  text-align: left;
  align-items: center;
  background-color: white;
  border-style: outset;
  padding-left: 100px;
  padding-right: 100px;
  padding-top: 5px;
  padding-bottom: 5px;
  font-family: "Open Sans Condensed", sans-serif;
  font-weight: 300;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.button {
  font-family: "Open Sans Condensed", sans-serif;
  color: rgb(234, 250, 146);
  padding: 8px 16px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 25px;
  margin: 4px 2px;
  transition-duration: 0.4s;
  cursor: pointer;
  border-radius: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.button1 {
  background-color: rgb(195, 233, 152);
  color: black;
  border: 2px solid #141b14;
}

.button1:hover {
  background-color: #4caf50;
  color: white;
}

.heading {
  font-size: 2rem;
  font-weight: 700;
}

.title {
  font-size: 3rem;
  font-weight: 700;
  font-family: "Open Sans Condensed", sans-serif;
  justify-content: center;
  text-align: center;
  background-color: rgb(227, 248, 221);
  color: rgb(58, 173, 58);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  border-style:outset;
}

.price {
  font-family: "Open Sans", sans-serif;
  font-size: 1.8rem;
  font-weight: 400;
}

.title .subtitle {
  font-size: 2rem;
  font-weight: 300;
  font-family: "Open Sans Condensed", sans-serif;
  justify-content: center;
  text-align: center;
  background-color: rgb(178, 243, 158);
  color: rgb(4, 41, 4);
}
