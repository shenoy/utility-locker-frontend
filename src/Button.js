import React from "react";

/* 
  Button Component
  @param {string} name
  @param {func} handleClick
  @returns button element
 */
const Button = ({ name, handleClick }) => {
  return (
    <input
      className="button button1"
      type="button"
      value={name}
      onClick={handleClick}
    />
  );
};

export default Button;
