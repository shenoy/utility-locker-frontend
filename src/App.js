import React from "react";
import "./App.css";
import {
  sortByPrice,
  sortByPriceDesc,
  fetchProducts,
  handleChangeDepartment,
  handleChangeType,
} from "./utils";
import Button from "./Button";
import { baseUrl } from "./utils";
class App extends React.Component {
  //state = { products: [] };

  constructor(props) {
    super(props);
    this.state = {
      products: [
        [
          {
            id: "SGN-2",
            name: "Samsung Galaxy Note 2",
            description: "Samsung tablet computer",
            price: {
              value: 299.99,
              currency: "GBP",
            },
            type: "Electrical",
            department: "Computing",
            weight: "569g",
          },
        ],
      ],
    };
    this.setState = this.setState.bind(this);
    this.handleChangeDepartment = this.handleChangeDepartment.bind(this);
  }

  componentDidMount() {
    this.fetchProducts();
  }

  fetchProducts = async () => {
    const res = await fetchProducts();
    return res;
  };

  handleChangeDepartment = async (props) => {
    const res = await handleChangeDepartment(props);
    this.setState({
      products: [res],
    });
  };

  handleChangeType = async (props) => {
    const res = await handleChangeType(props);
    this.setState({
      products: [res],
    });
  };

  sortByPrice = async () => {
    const res = await sortByPrice();
    this.setState({
      products: [res],
    });
  };

  sortByPriceDesc = async () => {
    const res = await sortByPriceDesc();
    this.setState({
      products: [res],
    });
  };

  render() {
    if (this.state.products.length > 0) {
      return (
        <div className="App">
          <div>
            <div className="title">
              <span>Utility Locker</span>
              <div className="subtitle">Range of Mobile Products</div>
            </div>

            <div className="sort">
              Sort by Price :
              <Button name="Low to High" handleClick={this.sortByPrice} />
              <Button name="High to Low" handleClick={this.sortByPriceDesc} />
            </div>
            <div className="group">
              Filter by Department :
              <Button
                name="Computing"
                handleClick={() => this.handleChangeDepartment("Computing")}
              />
              <Button
                name="Cookware"
                handleClick={() => this.handleChangeDepartment("Cookware")}
              />
              <Button
                name="DIY"
                handleClick={() => this.handleChangeDepartment("DIY")}
              />
              <Button
                name="Books and Stationery"
                handleClick={() =>
                  this.handleChangeDepartment("Books and Stationery")
                }
              />
              <Button
                name="Shoes"
                handleClick={() => this.handleChangeDepartment("Shoes")}
              />
              <Button
                name="Home"
                handleClick={() => this.handleChangeDepartment("Home")}
              />
              <Button
                name="Photography and Art"
                handleClick={() =>
                  this.handleChangeDepartment("Photography and Art")
                }
              />
              <Button
                name="Entertainment"
                handleClick={() => this.handleChangeDepartment("Entertainment")}
              />
            </div>
            <div className="group">
              Filter by Type :
              <Button
                name="Electrical"
                handleClick={() => this.handleChangeType("Electrical")}
              />
              <Button
                name="Voucher"
                handleClick={() => this.handleChangeType("Voucher")}
              />
              <Button
                name="Book"
                handleClick={() => this.handleChangeType("Book")}
              />
              <Button
                name="Ceramics"
                handleClick={() => this.handleChangeType("Ceramics")}
              />
              <Button
                name="Running"
                handleClick={() => this.handleChangeType("Running")}
              />
              <Button
                name="Electronics"
                handleClick={() => this.handleChangeType("Electronics")}
              />
            </div>

            {this.state.products[0].map((el, index) => (
              <div key={index}>
                <div className="item">
                  <div className="heading">{el.name}</div>
                  <div className="flex">
                    <div className="property">Description:</div>
                    &nbsp;<div>{el.description}</div>
                  </div>
                  <div className="flex">
                    <div className="property">Department:</div> &nbsp;
                    <div>{el.department}</div>
                  </div>

                  <div className="flex">
                    <div className="property">ID:</div>
                    &nbsp;<div>{el.id}</div>
                  </div>

                  <div className="flex">
                    <div className="property">Price:</div>
                    &nbsp;
                    <div className="price">Â£{el.price.value}</div>
                  </div>

                  <div className="flex">
                    <div className="property">Type:</div>
                    &nbsp;<div>{el.type}</div>
                  </div>

                  <div className="flex">
                    <div className="property">Weight:</div>
                    &nbsp;<div>{el.weight}</div>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
      );
    } else {
      return <h1>...loading</h1>;
    }
  }
}

export default App;
