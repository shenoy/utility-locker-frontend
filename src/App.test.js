import React from "react";
import ReactDOM from "react-dom";
import sinonChai from "sinon-chai";
import chai from "chai";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import {
  sortByPrice,
  sortByPriceDesc,
  fetchProducts,
  handleChangeType,
  handleChangeDepartment,
} from "./utils";
import App from "./App";
import Button from "./Button";
import Adapter from "enzyme-adapter-react-16";

chai.use(sinonChai);
configure({ adapter: new Adapter() });

describe("Sample Test", () => {
  it("test to be true", () => {
    const foo = true;
    expect(foo).toBe(true);
  });

  it("test to be false", () => {
    const foo = false;
    expect(foo).toBe(false);
  });
  it("knows that 2 and 2 make 4", () => {
    expect(2 + 2).toEqual(4);
  });
});

describe("Integration testing", () => {
  it("App renders without crashing ", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
  });

  it("App has an initial state", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state("products").length).toEqual(1);
  });
  it("App code matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("includes 1 div with class title", async () => {
    const wrapper = shallow(<App />);
    const span = wrapper.find("span");
    const result = span.text();
    await (() => expect(result).toEqual("Utility Locker"));
  });

  it("renders 2 group divs- group by Department and group by Type", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("div.group").length).toEqual(2);
  });

  it("renders a sort by Price div", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("div.sort").length).toEqual(1);
  });

  it("should render title", () => {
    const wrapper = shallow(<App />);
    expect(
      wrapper.containsAllMatchingElements([
        <div className="title">
          <span>Utility Locker</span>
          <div className="subtitle">Range of Mobile Products</div>
        </div>,
      ])
    ).toBe(true);
  });

  it("contains 16 <Button /> components", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Button).length).toEqual(16);
  });
});

describe("Button Component Testing", () => {
  it("button should be defined", () => {
    expect(Button).toBeDefined();
  });
  it("button should render correctly", () => {
    const tree = shallow(<Button name="button test" />);
    expect(tree).toMatchSnapshot();
  });
  it("should call mock function when button is clicked", () => {
    const mockFn = jest.fn();
    const tree = shallow(<Button name="button test" handleClick={mockFn} />);
    tree.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });

  it("should return the topmost element tag of the button", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.name()).toEqual("input");
  });
});

describe("Unit Testing App class Methods", () => {
  it("sortByPrice should return products sorted by price with lowest first", async () => {
    const data = await sortByPrice();
    expect(data[0].name).toEqual("Raspberry Pi Zero"); //lowest priced product
  });

  it("sortByPriceDesc should return products sorted by price with highest first", async () => {
    const data = await sortByPriceDesc();
    expect(data[0].name).toEqual("Samsung Galaxy Note 2"); //highest priced product
  });

  it("fetchProducts should return all products", async () => {
    const data = await fetchProducts();
    expect(data.length).toEqual(15); // count of all the products
  });

  it("handleChangeDepartment should fetch products of only the requested Department", async () => {
    const data = await handleChangeDepartment("Computing");
    expect(data[0].department).toEqual("Computing");
  });

  it("handleChangeType should fetch products of only the requested Type", async () => {
    const data = await handleChangeType("Electrical");
    expect(data[0].type).toEqual("Electrical");
  });
});
