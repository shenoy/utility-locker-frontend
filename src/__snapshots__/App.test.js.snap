// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Testing button should render correctly 1`] = `ShallowWrapper {}`;

exports[`Integration testing App code matches the snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="title"
    >
      <span>
        Utility Locker
      </span>
      <div
        className="subtitle"
      >
        Range of Mobile Products
      </div>
    </div>
    <div
      className="sort"
    >
      Sort by Price :
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Low to High"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="High to Low"
      />
    </div>
    <div
      className="group"
    >
      Filter by Department :
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Computing"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Cookware"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="DIY"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Books and Stationery"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Shoes"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Home"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Photography and Art"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Entertainment"
      />
    </div>
    <div
      className="group"
    >
      Filter by Type :
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Electrical"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Voucher"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Book"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Ceramics"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Running"
      />
      <input
        className="button button1"
        onClick={[Function]}
        type="button"
        value="Electronics"
      />
    </div>
    <div>
      <div
        className="item"
      >
        <div
          className="heading"
        >
          Samsung Galaxy Note 2
        </div>
        <div
          className="flex"
        >
          <div
            className="property"
          >
            Description:
          </div>
           
          <div>
            Samsung tablet computer
          </div>
        </div>
        <div
          className="flex"
        >
          <div
            className="property"
          >
            Department:
          </div>
            
          <div>
            Computing
          </div>
        </div>
        <div
          className="flex"
        >
          <div
            className="property"
          >
            ID:
          </div>
           
          <div>
            SGN-2
          </div>
        </div>
        <div
          className="flex"
        >
          <div
            className="property"
          >
            Price:
          </div>
           
          <div
            className="price"
          >
            £
            299.99
          </div>
        </div>
        <div
          className="flex"
        >
          <div
            className="property"
          >
            Type:
          </div>
           
          <div>
            Electrical
          </div>
        </div>
        <div
          className="flex"
        >
          <div
            className="property"
          >
            Weight:
          </div>
           
          <div>
            569g
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
